### PR TITLE
Add an upgrade test from EventMesh to the new Kyma eventing

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Add an upgrade test from EventMesh to the new Kyma Eventing.

**Hints for reviewers**
- The chart changes are done in another [PR](https://github.com/kyma-project/kyma/pull/10793) which also enables the new Eventing components.
- Since this is a special situation, both the **EventMeshUpgradeTest** and **EventingUpgradeTest** are using the same namespace for creating and verifying Eventing resources.
- We had to copy the v1alpha1 Subscription package from the [eventing-controller](https://github.com/kyma-project/kyma/tree/release-1.19/components/eventing-controller/api/v1alpha1) because we cannot upgrade Kyma version to 1.19.0 because of Kubernetes version collision with Knative 0.12.0.
- **EventMeshUpgradeTest** `CreateResources` will be executed, but `TestResources` will be skipped.
- **EventingUpgradeTest** `CreateResources` will skipped, but `TestResources` will be executed.
- **EventMeshUpgradeTest** will be deleted in a follow-up ticket: https://github.com/kyma-project/kyma/issues/10866.

### CODE REVIEW

File: tests/end-to-end/upgrade/pkg/tests/eventmesh/eventing.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		k8sClient:     k8sClient,
+		dynamicClient: dynamicClient,
+
+		// test config same as the EventMesh test config
+		eventTypeVersion:    "v1",
+		eventType:           defaultName,
+		subscriberName:      defaultName,
+		applicationName:     defaultName,
+		subscriptionName:    defaultName,
+		serviceInstanceName: defaultName,
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply tests/end-to-end/upgrade/pkg/tests/eventmesh/eventing.go:51]: All are needed here:

- func (e EventingUpgradeTest) waitForSubscriptionReady() error
- func (e EventingUpgradeTest) waitForSubscriber() error
- func (e EventingUpgradeTest) publishTestEvent() error
- func (e EventingUpgradeTest) checkEvent() error

File: tests/end-to-end/upgrade/pkg/tests/eventmesh/eventing.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	dynamicClient dynamic.Interface
+
+	// test config
+	namespace           string
+	eventType           string
+	subscriberName      string
+	applicationName     string
+	eventTypeVersion    string
+	subscriptionName    string
+	serviceInstanceName string
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply tests/end-to-end/upgrade/pkg/tests/eventmesh/eventing.go:37]: All are needed here:

- func (e EventingUpgradeTest) waitForSubscriptionReady() error
- func (e EventingUpgradeTest) waitForSubscriber() error
- func (e EventingUpgradeTest) publishTestEvent() error
- func (e EventingUpgradeTest) checkEvent() error

File: tests/end-to-end/upgrade/pkg/tests/eventmesh/eventing.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (k15r) [reply tests/end-to-end/upgrade/pkg/tests/eventmesh/eventing.go:51]: the serviceInstanceName is never used, but this should not block this PR

[the remark this answers is not in the document]
Comment (marcobebway) [reply tests/end-to-end/upgrade/pkg/tests/eventmesh/eventing.go:51]: Right, we will have a follow ticket for cleanup here: https://github.com/kyma-project/kyma/issues/10866.

**Review by k15r — COMMENTED** [review]
lgtm

**Comment by marcobebway** [comment]
Sample successful run: https://status.build.kyma-project.io/view/gcs/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/10793/pre-master-kyma-gke-upgrade/1371229074663411712.

**Review by suleymanakbas91 — APPROVED** [review]

**Pull request merged**